lib/src/model.dart

```diff
@@ -2759,53 +2998,78 @@ enum PackageWarning {
+  final String warningName;
+  final String shortHelp;
+  List<String> longHelp;
+  final PackageWarning warning;
+
+  PackageWarningHelpText(this.warning, this.warningName, this.shortHelp,
+      [this.longHelp]) {
+    if (this.longHelp == null) this.longHelp = [];
+  }
+}
+
 /// Provides description text and command line flags for warnings.
 /// TODO(jcollins-g): Actually use this for command line flags.
-Map<PackageWarning, List<String>> packageWarningText = {
-  PackageWarning.ambiguousDocReference: [
-    "ambiguous-doc-reference",
-    "A comment reference could refer to two or more different objects"
-  ],
-  PackageWarning.ambiguousReexport: [
-    "ambiguous-reexport",
-    "A symbol is exported from private to public in more than one place and dartdoc is forced to guess which one is canonical"
-  ],
-  PackageWarning.noCanonicalFound: [
-    "no-canonical-found",
-    "A symbol is part of the public interface for this package, but no library documented with this package documents it so dartdoc can not link to it"
-  ],
-  PackageWarning.noLibraryLevelDocs: [
-    "no-library-level-docs",
-    "There are no library level docs for this library"
-  ],
-  PackageWarning.categoryOrderGivesMissingPackageName: [
-    "category-order-gives-missing-package-name",
-    "The category-order flag on the command line was given the name of a nonexistent package"
-  ],
-  PackageWarning.unresolvedDocReference: [
-    "unresolved-doc-reference",
-    "A comment reference could not be found in parameters, enclosing class, enclosing library, or at the top level of any documented library with the package"
-  ],
-  PackageWarning.brokenLink: [
-    "brokenLink",
-    "Dartdoc generated a link to a non-existent file"
-  ],
-  PackageWarning.orphanedFile: [
-    "orphanedFile",
-    "Dartdoc generated files that are unreachable from the index"
-  ],
-  PackageWarning.unknownFile: [
-    "unknownFile",
-    "A leftover file exists in the tree that dartdoc did not write in this pass"
-  ],
-  PackageWarning.missingFromSearchIndex: [
-    "missingFromSearchIndex",
-    "A file generated by dartdoc is not present in the generated index.json"
-  ],
-  PackageWarning.typeAsHtml: [
-    "typeAsHtml",
-    "Use of <> in a comment for type parameters is being treated as HTML by markdown"
-  ],
+Map<PackageWarning, PackageWarningHelpText> packageWarningText = {
+  PackageWarning.ambiguousDocReference: new PackageWarningHelpText(
+      PackageWarning.ambiguousDocReference,
+      "ambiguous-doc-reference",
+      "A comment reference could refer to two or more different objects"),
+  PackageWarning.ambiguousReexport: new PackageWarningHelpText(
+      PackageWarning.ambiguousReexport,
+      "ambiguous-reexport",
+      "A symbol is exported from private to public in more than one library and dartdoc can not determine which one is canonical",
+      [
+        "Use {@canonicalFor @@name@@} in the desired library's documentation to resolve",
```

Comment:
I picked `@canonicalFor` since the tag is to appear in the library which is canonical for the symbol given as a parameter.  It'll look like this:

```
/// Some docs for foo.
/// {@canonicalFor bar.Baz}
library foo;
```

Since we're not specifying a location as a parameter to the tag I tend to drop out options 2-4.  I'm not sure what the first one would mean in this context.  The last one seems to add unnecessary information (AFAIK there are no similar comment tags for other tools).